src/if_python.c

```diff
@@ -928,7 +928,12 @@ Python_Init(void)
 #endif
 
 #ifdef PYTHON_HOME
+# ifdef DYNAMIC_PYTHON
+	if (mch_getenv((char_u *)"PYTHONHOME") == NULL)
```

Comment:
I think that you should only guard the condition with an additional `#ifdef` and not repeat `Py_SetPythonHome` call.